metrics.nim

```diff
@@ -170,8 +178,12 @@ proc newCounterMetrics*(name: string, labels, labelValues: Labels): seq[Metric]
             value: getTime().toMilliseconds().float64),
     ]
 
-proc newCounter*(name: string, help: string, labels: Labels = @[], registry = defaultRegistry): Counter =
-  when defined(metrics):
+  proc validateCounterLabelValues(counter: Counter, labelValues: Labels): Labels =
+    result = validateLabelValues(counter, labelValues)
+    if result notin counter.metrics:
+      counter.metrics[result] = newCounterMetrics(counter.name, counter.labels, result)
+
+  proc newCounterImpl*(name: string, help: string, labels: Labels = @[], registry = defaultRegistry): Counter =
```

Comment:
A public template is allowed to use a private proc, so you don't need to export this here.

Reply:
It doesn't work.